packages/core/types/src/modules/entity-service/params/filters/index.ts

```diff
@@ -9,34 +9,78 @@ export { Operator };
 type IDKey = 'id';
 
 /**
- * Filter representation for nested attributes
+ * Generic object notation for filters.
+ * @template TSchemaUID The type of the schema UID for the object notation.
  */
-type NestedAttributeCondition<
-  TSchemaUID extends Common.UID.Schema,
-  TAttributeName extends Attribute.GetKeys<TSchemaUID>
-> = ObjectNotation<
-  Utils.Guard.Never<Attribute.GetTarget<TSchemaUID, TAttributeName>, Common.UID.Schema>
+export type Any<TSchemaUID extends Common.UID.Schema> = ObjectNotation<TSchemaUID>;
+
+/**
+ * Type that unites root-level operators and attributes filtering for a specific schema query.
+ *
+ * It is used to define the structure of filters objects in a specific schema.
+ * @template TSchemaUID The UID of the schema defining the object notation.
+ */
+export type ObjectNotation<TSchemaUID extends Common.UID.Schema> =
+  RootLevelOperatorFiltering<TSchemaUID> & AttributesFiltering<TSchemaUID>;
+
+/**
+ * Object for root level operator filtering.
+ * @template TSchemaUID - The type of the schema UID.
+ */
+export type RootLevelOperatorFiltering<TSchemaUID extends Common.UID.Schema> = {
+  [TIter in Operator.Group]?: ObjectNotation<TSchemaUID>[];
+} & {
+  [TIter in Operator.Logical]?: ObjectNotation<TSchemaUID>;
+};
+
+/**
+ * Represents a type for filtering on attributes based on a given schema.
+ *  @template TSchemaUID - The UID of the schema.
+ */
+export type AttributesFiltering<TSchemaUID extends Common.UID.Schema> = Utils.Guard.Never<
+  // Combines filtering for scalar and nested attributes based on schema UID
+  ScalarAttributesFiltering<TSchemaUID> & NestedAttributeFiltering<TSchemaUID>,
+  // Abstract representation of the filter object tree in case we don't have access to the attributes' list
+  {
+    [TKey in string]?:
+      | AttributeCondition<TSchemaUID, never>
+      | NestedAttributeCondition<TSchemaUID, never>;
+  }
 >;
 
 /**
- * Filter representation for scalar attributes
+ * Definition of scalar attribute filtering for a given schema UID.
+ * @template TSchemaUID - The UID of the schema.
+ */
+export type ScalarAttributesFiltering<TSchemaUID extends Common.UID.Schema> =
+  AttributeUtils.GetScalarKeys<TSchemaUID> extends infer TKeys
+    ? TKeys extends AttributeUtils.GetScalarKeys<TSchemaUID>
+      ? IDFiltering & { [TKey in TKeys]?: AttributeCondition<TSchemaUID, TKey> }
+      : never
+    : never;
```

Comment:
```suggestion
export type ScalarAttributesFiltering<TSchemaUID extends Common.UID.Schema> = IDFiltering &
  AttributeUtils.GetScalarKeys<TSchemaUID> extends infer TKeys
    ? TKeys extends AttributeUtils.GetScalarKeys<TSchemaUID>
      ? { [TKey in TKeys]?: AttributeCondition<TSchemaUID, TKey> }
      : never
    : never;
```

Should this be like that ? Not sure that would work will with the never 🤔

Reply:
It'll factorize to the same thing. In both examples, if it cannot infer the scalar keys it'll end up being never for the resulting type.

As for the form itself, I find that less confusing since you understand the above explanation at first glance. I'm not against yours though. Whatever feels the most elegant & readable :shrug:

Reply:
If there is no bug no need to change I was thinking ID should be there regardless of the scalar check :)

Reply:
We might consider that later, for now I'll keep the same behavior as before regarding the ID filters and just fix the issue at hand (trying to avoid adding side effect to the PR)

Reply:
Thanks for the suggestion though